tests/test_workflow_api.py

```diff
@@ -60,3 +59,44 @@ async def test_workflow_post(
 
             assert db_workflow.name == "My Workflow"
             assert db_workflow.project_id == id
+
+
+async def test_workflow_delete(
+    db, client, MockCurrentUser, project_factory, task_factory
+):
+    """
+    GIVEN a Workflow with two Tasks
+    WHEN the delete endpoint is called
+    THEN the Workflow and its associated WorkflowTasks
+        are removed from the db
+    """
+    async with MockCurrentUser(persist=True) as user:
+        project = await project_factory(user)
+        p_id = project.id
+        workflow = {
+            "name": "My Workflow",
+            "project_id": p_id,
+        }
+
+        res = await client.post(
+            "api/v1/workflow/",
+            json=workflow,
+        )
+        wf = (await db.execute(select(Workflow))).scalars().one()
+        t0 = await task_factory()
+        t1 = await task_factory()
+        await wf.insert_task(t0, db=db)
+        await wf.insert_task(t1, db=db)
+
+        assert len((await db.execute(select(Workflow))).scalars().all()) == 1
+        assert (
+            len((await db.execute(select(WorkflowTask))).scalars().all()) == 2
+        )
```

Comment:
it would be better to query only the `WorkflowTask`s belonging to the `Workflow` you are dealing with. For instance, this would allow to also test that other `WorkflowTask`s are not touched by the delete.